chainset.go

```diff
@@ -0,0 +1,136 @@
+package ibctest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/ory/dockertest/v3"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"golang.org/x/sync/errgroup"
+)
+
+// chainSet is an ordered collection of ibc.Chain,
+// to group methods that apply actions against all chains in the set.
+//
+// The main purpose of the chainSet is to unify test setup when working with any number of chains.
+type chainSet []ibc.Chain
+
+// Initialize concurrently calls Initialize against each chain in the set.
+// Each chain may run a docker pull command,
+// so with a cold image cache, running concurrently may save some time.
+func (cs chainSet) Initialize(testName string, homeDir string, pool *dockertest.Pool, networkID string) error {
+	var eg errgroup.Group
+
+	for _, c := range cs {
+		c := c
+		eg.Go(func() error {
```

Comment:
This pattern is interesting to me. I almost always avoid shadowing variables and package names in Go. I tend to use: 

```go
for i := range cs {
   c := cs[i]
```

Both have the same result. 

Just an observation, no need to change.

Reply:
[Effective Go](https://go.dev/doc/effective_go) states:

> It may seem odd to write `req := req` but it's legal and idiomatic in Go to do this. You get a fresh version of the variable with the same name, deliberately shadowing the loop variable locally but unique to each goroutine.

I think the _idiomatic_ detail specifically applies when it is the first statement inside the loop.